src/main/java/com/_up/megastore/controllers/implementations/ProductController.java

```diff
@@ -20,5 +25,11 @@ public ProductController(IProductService productService) {
     public ProductResponse saveProduct(CreateProductRequest createProductRequest, MultipartFile multipartFile) {
         return productService.saveProduct(createProductRequest, multipartFile);
     }
+    @Override
+    public ProductResponse updateProduct(@RequestBody Product product){
+       return productService.updateProduct(UpdateProductRequest updateProductRequest, MultipartFile multipartFile);
+
+    }
```

Comment:
Acá se supone que estás implementando el método que definiste en la interfas IProductController, pero no coinciden los parámetros qeu definis en la interfaz, esto incluso te debería dar error.

Además, estás llamando al método updateProduct del productService pasándole el tipo de dato de los parámetros. Cuando llamás a un método solo tenes que pasarle el parámetro, no el tipo de dato

```suggestion
    @Override
    public ProductResponse updateProduct(UpdateProductRequest updateProductRequest, MultipartFile multipartFile){
       return productService.updateProduct(updateProductRequest, multipartFile);

    }
```